### ISSUE TITLE
[MPMD-91] pmd does not report correct file name for unneccessary imports

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-pmd-plugin#457]
**[Sean Bridges](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sgbridges)** opened **[MPMD-91](https://issues.apache.org/jira/browse/MPMD-91?redirect=false)** and commented

If I have a class Foo in package com.corp, with unused imports, `mvn pmd:pmd pmd:check` reports,

```
PMD Failure: com.corp.5 Rule:UnusedImports Priority:4 Avoid unused imports such as 'com.corp.bar.Bar'.
```

The report does not give the class name (Foo), merely the package and line number.  Looking in `pmd.xml` we have,

```xml
<file name="<filename>">
<violation beginline="5" endline="5" begincolumn="1" endcolumn="78" rule="UnusedImports" ruleset="Import Statement Rules" package="com.corp" externalInfoUrl="http://pmd.sourceforge.net/rules/imports.html#UnusedImports" priority="4">
Avoid unused imports such as 'com.corp.bar.Bar'
</violation>
</file>
```

pmd correctly does not report a class attribute in the violation element, since the package imports are not associated with a class.

The mvn output doesn't tell me which file has the error.  If it tells me the class name, I can find the file, but with the output above all I know is that some file in the package com.corp has a pmd error.  If the class can't be found, it would be nice to indicate the file name somehow.

It may be better to always show the filename rather than class name in the mvn output.  The filename is what I am interested in seeing, and will always be there.

---

**Affects:** 2.4

**Issue Links:**
- [MPMD-98](https://issues.apache.org/jira/browse/MPMD-98) The plugin console output is inconsistent
 (_**"is duplicated by"**_)
- [MPMD-109](https://issues.apache.org/jira/browse/MPMD-109) Class name not printed in pmd:check output
 (_**"is duplicated by"**_)
